src/base/utils/misc.cpp

```diff
@@ -634,3 +637,35 @@ QSize Utils::Misc::smallIconSize()
     return QSize(s, s);
 }
 #endif
+
+QString Utils::Misc::osName()
+{
+    // static initialization for usage in signal handler
+    static const QString name =
+#ifdef QBT_USES_QT5
+    QString("%1 %2 %3")
+    .arg(QSysInfo::prettyProductName())
+    .arg(QSysInfo::kernelVersion())
+    .arg(QSysInfo::currentCpuArchitecture());
+#else
+    "<Input OS name here>";
+#endif
+    return name;
+}
+
+QString Utils::Misc::boostVersionString()
+{
+    // static initialization for usage in signal handler
+    const int major = BOOST_VERSION / 100000;
+    const int minor = ((BOOST_VERSION / 100) % 1000);
+    const int subMin = BOOST_VERSION % 100;
+    static const QString ver = QString("%1.%2.%3").arg(major).arg(minor).arg(subMin);
```

Comment:
It's not my desire. Otherwise, they are computed each time you call the function, it is worthless.